django/db/backends/utils.py

```diff
@@ -135,8 +135,10 @@ def typecast_timestamp(s):  # does NOT store time zone information
     # "2005-07-29 09:56:00-05"
     if not s:
         return None
-    if ' ' not in s:
+    if ':' not in s:
         return typecast_date(s)
+    if '-' not in s:
+        return typecast_time(s)
```

Comment:
I agree that this function the and the SQLite3 backend ones depending on it could benefit from a refactor.